File: Projects/Example/Example/Code/Pages/17-XZNavigationController/Example17Model.m
//
//  Example17Model.m
//  Example
//
//  Created by 徐臻 on 2026/3/20.
//

#import "Example17Model.h"
#import "Example17Configuration.h"

@implementation Example17Model

- (instancetype)init {
    return [self initWithConfiguration:[Example17Configuration new]];
}

- (instancetype)initWithConfiguration:(Example17Configuration *)configuration {
    self = [super init];
    if (self) {
        _current = configuration;
        _next = [[Example17Configuration alloc] init];
    }
    return self;
}

@end

File: Projects/Example/Example/Code/Pages/17-XZNavigationController/Model/Example17Model.swift
//
//  Example17Model.swift
//  Example
//
//  Created by Xezun on 2026/3/21.
//

import UIKit
import XZKit

@mocoa
class Example17Model: NSObject, XZMocoaModel {
    
    dynamic var current : Example17Configuration
    dynamic var next    : Example17Configuration
    
    init(current: Example17Configuration = .init()) {
        self.current = current
        self.next = Example17Configuration.init()
        super.init()
    }
    
}

@objc class Example17Configuration: NSObject {
    @objc dynamic var isHidden = false
    @objc dynamic var isTranslucent = true
    @objc dynamic var prefersLargeTitles = false
}
